Source/FieldSolver/ElectrostaticSolver.cpp

```diff
@@ -62,14 +62,50 @@ WarpX::ComputeSpaceChargeField (bool const reset_fields)
 
     if (do_electrostatic == ElectrostaticSolverAlgo::LabFrame) {
         AddSpaceChargeFieldLabFrame();
-    } else {
+    }
+    else {
+        bool calculate_background_field = false;
         // Loop over the species and add their space-charge contribution to E and B
         for (int ispecies=0; ispecies<mypc->nSpecies(); ispecies++){
             WarpXParticleContainer& species = mypc->GetParticleContainer(ispecies);
             if (species.initialize_self_fields ||
                 (do_electrostatic == ElectrostaticSolverAlgo::Relativistic)) {
                 AddSpaceChargeField(species);
+                calculate_background_field = true;
+            }
+        }
+
+        if (calculate_background_field){
```

Comment:
To be maybe discussed: I would suggest:
```suggestion
        if (do_electrostatic == ElectrostaticSolverAlgo::Relativistic){
```
(and then remove the variable `calculate_background_field` above)

This essentially results in the E field from the boundary potential to **not** be calculated when we use `initialize_self_fields`. This is because `initialize_self_fields` is typically used to initialize the fields on the grid at `t=0`, when using the **Maxwell solver**. But, when using the Maxwell solver, the E field from `phi` at the boundaries is in principle ignored. In fact, if we add this (static) field at `t=0`, I am not sure it will remain unchanged when applying the Maxwell solver at each timestep.